app/workers/builds.php

```diff
@@ -246,24 +248,20 @@ protected function buildDeployment(Document $project, Document $function, Docume
                 channels: $target['channels'],
                 roles: $target['roles']
             );
-
-            /** Update usage stats */
-            if (App::getEnv('_APP_USAGE_STATS', 'enabled') === 'enabled') {
-                $statsd = $register->get('statsd');
-                $usage = new Stats($statsd);
-                $usage
-                    ->setParam('projectInternalId', $project->getInternalId())
-                    ->setParam('projectId', $project->getId())
-                    ->setParam('functionId', $function->getId())
-                    ->setParam('builds.{scope}.compute', 1)
-                    ->setParam('buildStatus', $build->getAttribute('status', ''))
-                    ->setParam('buildTime', $build->getAttribute('duration'))
-                    ->setParam('buildSize', $build->getAttribute('size'))
-                    ->setParam('networkRequestSize', 0)
-                    ->setParam('networkResponseSize', 0)
-                    ->submit();
-            }
         }
+
+        /** Trigger usage queue */
+        $this
+            ->getUsageQueue()
+            ->setProject($project)
+            ->addMetric("builds", 1) // per project
+            ->addMetric("builds.storage", $build->getAttribute('size', 0))
+            ->addMetric("builds.compute", $build->getAttribute('duration', 0))
+            ->addMetric("{$function->getInternalId()}" . ".builds", 1) // per function
+            ->addMetric("{$function->getInternalId()}" . ".builds.storage", $build->getAttribute('size', 0))
+            ->addMetric("{$function->getInternalId()}" . ".builds.compute", $build->getAttribute('duration', 0))
```

Comment:
We need to define these metrics as constants somewhere. Else they will get very hard to manage

BUILDS_COUNT = 'builds';
BUILDS_STORAGE = 'builds.storage';
BUILDS_COMPUTE = 'builds.compute';


FUNCTION_BUILDS_COUNT = '{functionInternalId}.builds';

and so on..

Reply:
I also see that we're missing the `buildStatus` metric.